### PR TITLE
Log-7549 and LOG-7550 Linux Agent Stories

### PR DESCRIPTION
**Pull request by pclune-r7**
Changes in this PR for Linux Agent

_api_mock.py_ 
- with new host and logs
- get 'loglist' response updated for new hosts/logs
- put IngestionHandler updated for new logs
- POST request handling is now dynamic for core multilog test cases   

_le.py_
- LE updated to handle new paramter --multilog
- Function added to verify pathname (from command line; config file or server side) against rules per story
- Function added for display of user prompt as per story requirment
- Function added for handling request to follow for 'multilog' option
- Text Added for Usage of Multilog
- Added new debug command --debug-multilog
- FollowMultiple object instantiated when a 'Multiple:' prefix detected in filename 
- FollowMultiple object creates a Follower object for each file found that matches pattern (where wildcard is present)
- For shutdown, individual FollowMultiple object will first 'close' all follower objects it is holding 
- FollowMultiple class will be responsible for 'dynamic' tracking of files and for managing the limitation on max number of files followed

_help.sh_
- updated to add new --mutilog option

_README.md_
- updated to include description of usage and example for --multilog    

_Test Files Added_
- multilog_configuration.sh
- multilog_dynamic.sh
- multilog_pathname_verification.sh
- multilog_workflow.sh

_test.sh_
- added new variable for TMP dictionary use on POST urlendcoding

_vars_
- updated for new HOST_MULTILOG_KEY

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,6 +67,7 @@ How to use
 	--suppress-ssl    do not use SSL with API server
 	--yes	          always respond yes
 	--pull-server-side-config=False do not use server-side config for following files
+	--multilog        option used with directory wildcard (restricted behaviour - see below for details)
```
Comment (vilda-r7) [inline README.md]: This option is for follow command only - I would expect it under --name and --type

File: src/le.py [hunk 2]
```diff
@@ -80,6 +80,21 @@
     "http": "803fe7ba-bd2e-44bd-8ee7-f02fa253ef5f",
 }
 
+# '--multilog' option related
+# max number of files that are allowed to be followed at any one time
```
Comment (vilda-r7) [inline src/le.py]: Consistency - can you capitalize first letters in comments

File: src/le.py [hunk 3]
```diff
@@ -1461,21 +1489,117 @@ def cancel(self):
...
+        # fail if a symbolic link
+        if os.path.islink(candidate):
+            return False
+        # fail if not a regular file (passes links!)
+        if not os.path.isfile(candidate):
+            return False
+        return True
+    
+    def _append_followers(self, add_files):
+
```
Comment (vilda-r7) [inline src/le.py]: Avoid empty lines

File: src/le.py [hunk 4]
```diff
@@ -1461,21 +1489,117 @@ def cancel(self):
...
+                follower = Follower(filename, self.entry_filter, self.entry_formatter, self.entry_identifier, self.transport, True)
+                self._followers.append(follower)
+                if config.debug_multilog:
+                    print >> sys.stderr, "Number of followers increased to: %s " %len(self._followers)
+            else:
+                log.debug("Warning: Allowed maximum of files that can be followed reached")
+                break
+                        
+    def _remove_followers(self, removed_files):
+
```
Comment (vilda-r7) [inline src/le.py]: Avoid this empty line

File: src/le.py [hunk 5]
```diff
@@ -1461,21 +1489,117 @@ def cancel(self):
...
+                                          
 
 class Follower(object):
 
     """
     The follower keeps an eye on the file specified and sends new events to the
     logentries infrastructure.  """
 
-    def __init__(self, name, entry_filter, entry_formatter, entry_identifier, transport):
+    def __init__(self, name, entry_filter, entry_formatter, entry_identifier, transport, disable_glob=False):
```
Comment (vilda-r7) [inline src/le.py]: Line too long?

File: README.md (hunk 1 above)
Comment (pclune-r7) [inline README.md]: will do

File: src/le.py (hunk 3 above)
Comment (pclune-r7) [inline src/le.py]: will do

File: src/le.py (hunk 2 above)
Comment (pclune-r7) [inline src/le.py]: will do

File: src/le.py (hunk 4 above)
Comment (pclune-r7) [inline src/le.py]: will do

File: src/le.py (hunk 5 above)
Comment (pclune-r7) [inline src/le.py]: okay - will change it along with the _init_ for FollowMultilog

**Pull request merged**